detection_rules/devtools.py

```diff
@@ -201,3 +204,223 @@ def license_check(ctx):
                 click.echo(relative_path, err=True)
 
     ctx.exit(int(failed))
+
+
+@dev_group.group('gh-release')
+def gh_release_group():
+    """Commands to manage GitHub releases."""
+
+
+@gh_release_group.command('create-ml')
+@click.argument('directory', type=click.Path(dir_okay=True, file_okay=False))
+@click.option('--gh-token', '-t', default=getdefault('gh_token'))
+@click.option('--repo', '-r', default='elastic/detection-rules', help='GitHub owner/repo')
+@click.option('--release-name', '-n', required=True, help='Name of release')
+@click.option('--description', '-d', help='Description of release to append to default message')
+@click.pass_context
+def create_ml_release(ctx, directory, gh_token, repo, release_name, description):
+    """Create a GitHub release."""
+    import re
+    import requests
+
+    pattern = r'(ML-DGA|ML-experimental-detections)-20\d\d[0-1]\d[0-3]\d-\d'
+    assert re.match(pattern, release_name), f'release name must match pattern: {pattern}'
+    assert Path(directory).name == release_name, f'directory name must match release name: {release_name}'
+
+    try:
+        from github import Github
+    except ImportError as e:
+        Github = None  # noqa: N806  # for type hinting
+        client_error('Missing PyGithub - try running `pip install -r requirements-dev.txt`', e)
+
+    gh_token = gh_token or click.prompt('GitHub token', hide_input=True)
+    client = Github(gh_token)
+    gh_repo = client.get_repo(repo)
+
+    # validate tag name is increment by 1
+    name_parts = release_name.rsplit('-')
+    name_prefix = '-'.join(name_parts[:2])
+    version = int(name_parts[-1])
+    releases = gh_repo.get_releases()
+    max_ver = max([int(r.raw_data['name'].rsplit('-', maxsplit=1)[-1]) for r in releases
+                   if r.raw_data['name'].startswith(name_prefix)], default=0)
+
+    if version != (max_ver + 1):
+        client_error(f'Last release version was {max_ver}. Release name should end with version: {max_ver + 1}')
+
+    # validate files
+    if name_prefix == 'ML-DGA':
+        zipped_bundle, description_str = ctx.invoke(validate_ml_dga_asset, directory=directory)
+    else:
+        zipped_bundle, description_str = ctx.invoke(validate_ml_detections_asset, directory=directory)
+
+    click.confirm('Validation passed, verify output. Continue?')
+
+    if description:
+        description_str = f'{description_str}\n\n----\n\n{description}'
+
+    release = gh_repo.create_git_release(name=release_name, tag=release_name, message=description_str)
+    zip_name = Path(zipped_bundle).name
+
+    # add zipped bundle as an asset to the release
+    with open(zipped_bundle, 'rb') as zipped_fo:
+        headers = {'content-type': 'application/zip'}
+        url = f'{release.upload_url[:-13]}?name={zip_name}&label={zip_name}'
```

Comment:
what are the last thirteen characters?
can you do `-len("foobar")` or something that's more obvious

Reply:
`{?name,label}`

I could do something like this

```suggestion
        suffix = '{?name,label}'
        url = release.upload_url.replace(suffix, f'?name={zip_name}&label={zip_name}')
```